pkg/registry/apis/provisioning/repository_controller.go

```diff
@@ -221,10 +225,18 @@ func (rc *RepositoryController) process(item *queueItem) error {
 	}
 
 	healthAge := time.Since(time.UnixMilli(obj.Status.Health.Checked))
+	syncAge := time.Since(time.UnixMilli(obj.Status.Sync.Finished))
+	syncInterval := time.Duration(obj.Spec.Sync.IntervalSeconds) * time.Second
+	tolerance := time.Second
+	isSyncJobPendingOrRunning := obj.Status.Sync.State == provisioning.JobStatePending || obj.Status.Sync.State == provisioning.JobStateWorking
+	shouldResync := syncAge >= (syncInterval-tolerance) && !isSyncJobPendingOrRunning
```

Comment:
syncInterval == 0 would also not trigger, correct

Reply:
yeah, this one is correct regardless of the default value. I don't want it to run.